lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/RandomSpec.hs

```diff
@@ -128,7 +184,57 @@ defMnemonic =
     [ "squirrel", "material", "silly", "twice", "direct", "slush"
     , "pistol", "razor", "become", "junk", "kingdom", "flee" ]
 
--- Get a private key from a hex string, without error checking.
+{-------------------------------------------------------------------------------
+                    Golden tests for Address derivation path
+-------------------------------------------------------------------------------}
+
+data DecodeDerivationPath = DecodeDerivationPath
+    { mnem :: [Text]
+    , addr :: ByteString
+    , accIndex :: Word32
+    , addrIndex :: Word32
+    } deriving (Generic, Show, Eq)
+
+decodeTest :: DecodeDerivationPath -> Expectation
+decodeTest DecodeDerivationPath{..} =
+    decoded `shouldBe` Right (Just (Index accIndex, Index addrIndex))
+  where
+    payload = unsafeDeserialiseFromBytes decodeAddressPayload $ b58decode addr
+    decoded = deserialise (decodeAddressDerivationPath rootXPub) payload
+    Key rootXPrv = generateKeyFromSeed (Passphrase seed) (Passphrase "")
+    rootXPub = Key (toXPub rootXPrv)
+    Right (Passphrase seed) = fromMnemonic @'[12] mnem
+
+-- Generated on mainnet Daedalus -- first address of the initial account.
+decodeTest1 :: DecodeDerivationPath
+decodeTest1 = DecodeDerivationPath
+    { mnem = addrMnemonic
+    , addr = "DdzFFzCqrhsznTSvu5VS2Arte6DbsvfGL2mhezwj4T8fvJqQ4C53RYc8nrNukdpfUfxz3R5ryZTcMtFZfdq4hVkPFHD1XV2dxY7AJEon"
+    , accIndex = 2147483648
+    , addrIndex = 2147483648
+    }
+
+-- Generated for mainnet, first address of an additional account.
+decodeTest2 :: DecodeDerivationPath
+decodeTest2 = DecodeDerivationPath
+    { mnem = addrMnemonic
+    , addr = "DdzFFzCqrht2qSNuod2j3HQdxQYu7ehMnHqPMK6ZCZc1oTBfFJFTaqMF62rzWsJWZhbrN15uA4Bsp6M7t5WkqfumdnjLjZ5xRk8szuCd"
+    , accIndex = 2694138340
+    , addrIndex = 2512821145
+    }
```

Comment:
It'd be good to also have golden tests for managing `testnet` addresses. The only change should be in the address attributes which contains an extra part.

Reply:
OK done. I needed to adjust the address attributes decoder used by the golden tests.